src/Hint/Extensions.hs

```diff
@@ -425,7 +427,7 @@ used EmptyCase = hasS f
     f _ = False
 used KindSignatures = hasT (un :: HsKind GhcPs)
 used BangPatterns = hasS isPBangPat ||^ hasS isStrictMatch
-used TemplateHaskell = hasS $ not . isQuasiQuoteSplice
+used TemplateHaskell = hasS (not . isQuasiQuoteSplice) ||^ hasS isTypedSplice
```

Comment:
Preempting potential review question:
Why not replace (not . isQuasiQuoteSplice) with `isUntypedSplice`?

<details>
  <summary>because that leads to bunch of test failures</summary>
  <pre>
TEST FAILURE (1 hints generated)
SRC: src/Hint/Extensions.hs:16:0
INPUT: {-# LANGUAGE TemplateHaskell #-} 
$(deriveNewtypes typeInfo)
OUTPUT: src/Hint/Extensions.hs:1:1-32: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Perhaps you should remove it.

WANTED: <failure>


TEST FAILURE (0 hints generated)
SRC: src/Hint/Extensions.hs:20:0
INPUT: {-# LANGUAGE QuasiQuotes, TemplateHaskell #-} 
f x = x + [e| x + 1 |] + [foo| x + 1 |]
WANTED: {-# LANGUAGE QuasiQuotes #-}


TEST FAILURE (1 hints generated)
SRC: src/Hint/Extensions.hs:24:0
INPUT: {-# LANGUAGE TemplateHaskell,EmptyDataDecls #-} 
$(fmap return $ dataD (return []) (mkName "Void") [] [] [])
OUTPUT: src/Hint/Extensions.hs:1:1-47: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell, EmptyDataDecls #-}
Perhaps you should remove it.

WANTED: <failure>


TEST FAILURE (1 hints generated)
SRC: src/Hint/Extensions.hs:75:0
INPUT: {-# LANGUAGE TemplateHaskell #-} 
foo
OUTPUT: src/Hint/Extensions.hs:1:1-32: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Perhaps you should remove it.

WANTED: <failure>
  </pre>
</details>